flow/diff_context.cc

```diff
@@ -12,21 +12,40 @@ DiffContext::DiffContext(SkISize frame_size,
                          PaintRegionMap& this_frame_paint_region_map,
                          const PaintRegionMap& last_frame_paint_region_map,
                          bool has_raster_cache)
-    : rects_(std::make_shared<std::vector<SkRect>>()),
+    : clip_tracker_(DisplayListMatrixClipTracker(kGiantRect, SkMatrix::I())),
+      rects_(std::make_shared<std::vector<SkRect>>()),
       frame_size_(frame_size),
       frame_device_pixel_ratio_(frame_device_pixel_ratio),
       this_frame_paint_region_map_(this_frame_paint_region_map),
       last_frame_paint_region_map_(last_frame_paint_region_map),
       has_raster_cache_(has_raster_cache) {}
 
+void DiffContext::MakeCurrentTransformIntegral() {
+  // TODO(knopp): This is duplicated from LayerStack. Maybe should be part of
+  // clip tracker?
```

Comment:
Another idea is to have a function define that takes an SkM44 ref and modifies it as necessary, paired with one that takes an SkMatrix ref - `updateTransform(MatrixUpdater44, MatrixUpdater33)`

For now, though, duping this one method seems to keep the concept of pixel snapping reasonably contained.

Reply:
And, finally, yeah, maybe we could just perform the same computations in the ClipTracker code, which would then dictate the pixel snapping policy for all clients.

Reply:
Perhaps move this method down to where the PushTransform method is (and the SetTransform method used to be) for conceptual grouping?

Reply:
I'm starting to lean towards putting this in ClipTracker since the use here involves unnecessary copying in/out of the matrix parameters. Also, if it is virtual then we don't need the test to see which variant to use.